hpx/lcos/future.hpp

```diff
@@ -1367,49 +1250,24 @@ namespace hpx { namespace lcos
 
         template <typename F>
         typename util::lazy_enable_if<
-            !hpx::traits::is_launch_policy<
-                typename std::decay<F>::type>::value &&
-            !hpx::traits::is_threads_executor<
-                typename std::decay<F>::type>::value &&
-            !hpx::traits::is_one_way_executor<
-                typename std::decay<F>::type>::value &&
-            !hpx::traits::is_two_way_executor<
-                typename std::decay<F>::type>::value
-          , hpx::traits::future_then_result<shared_future, F>
+            hpx::traits::is_invocable<F(shared_future&&)>::value,
```

Comment:
```suggestion
            hpx::traits::is_invocable<F, shared_future&&>::value,
```
(as pointed out by @K-ballo)